src/data/repositories/SignalDefaultRepository.ts

```diff
@@ -39,4 +43,34 @@ export class SignalDefaultRepository implements SignalRepository {
             }
         });
     }
+
+    private hasDeletePermission(signals: Signal[], currentOrgUnitId: string): FutureData<Signal[]> {
+        return apiToFuture(
+            this.api.models.organisationUnits.get({
+                fields: {
+                    id: true,
+                    level: true,
+                },
+            })
+        ).map(({ objects }) => {
+            const filteredOrgUnits = objects.filter(orgUnit =>
+                [...signals.map(signal => signal.orgUnit.id), currentOrgUnitId].includes(orgUnit.id)
+            );
+            const extendedSignals: Signal[] = [];
+            const currentOrgUnit = filteredOrgUnits.find(orgUnit => orgUnit.id === currentOrgUnitId);
+            signals.forEach(signal => {
+                const signalOrgUnitLevel = filteredOrgUnits.find(orgUnit => orgUnit.id === signal?.orgUnit.id)?.level;
+                if (
+                    signal.orgUnit.id === currentOrgUnitId ||
+                    (signalOrgUnitLevel && currentOrgUnit && signalOrgUnitLevel > currentOrgUnit?.level)
```

Comment:
Hi @davidbgomes i did not test this, so might be wrong, could you confirm? We are only checking that the signalOrgUnitLevel > currentOrgUnit?.level, but we are not checking if signal org unit and current org unit are in the same hierarchy. For example , if a user has permission for WPR , but the signal belongs to Burkina Faso. Even thought the level is higher, they don't belong to same hierarchy.

Reply:
Hello @9sneha-n ! I've just tested that scenario with a WPR orgUnit permission and the delete button is still disabled. I ran a few tests of the inverse, adding a signal of a orgUnit bellow the current one, and the deletion was enabled.